packages/salesforcedx-vscode-lwc/test/vscode-integration/activation.test.ts

```diff
@@ -0,0 +1,202 @@
+      fail('Test requires workspace folders');
+      return;
+    }
+
+    vscode.workspace.workspaceFolders.forEach(async workspaceRoot => {
+      const filePath = path.join(
+        workspaceRoot.uri.fsPath,
+        'force-app',
+        'main',
+        'default',
+        'lwc',
+        'demoLwcComponent',
+        'demoLwcComponent.js'
+      );
+
+      // Open an LWC js file
+      await vscode.workspace.openTextDocument(filePath);
+
+      assert.match(lwcExtension.isActive, true);
+    });
+  });
+
+  it('Language Server should start when an LWC css file is opened', async () => {
+    const lwcExtension: vscode.Extension<any> = vscode.extensions.getExtension(
+      'salesforce.salesforcedx-vscode-lwc'
+    ) as vscode.Extension<any>;
+
+    if (!vscode.workspace.workspaceFolders) {
+      fail('Test requires workspace folders');
+      return;
+    }
+
+    vscode.workspace.workspaceFolders.forEach(async workspaceRoot => {
+      const filePath = path.join(
+        workspaceRoot.uri.fsPath,
+        'force-app',
+        'main',
+        'default',
+        'lwc',
+        'demoLwcComponent',
+        'demoLwcComponent.css'
+      );
+
+      // Open an LWC css file
+      await vscode.workspace.openTextDocument(filePath);
+
+      assert.match(lwcExtension.isActive, true);
+    });
+  });
+
+  it('Language Server should start when an LWC html file is opened', async () => {
+    const lwcExtension: vscode.Extension<any> = vscode.extensions.getExtension(
+      'salesforce.salesforcedx-vscode-lwc'
+    ) as vscode.Extension<any>;
+
+    if (!vscode.workspace.workspaceFolders) {
+      fail('Test requires workspace folders');
+      return;
+    }
+
+    vscode.workspace.workspaceFolders.forEach(async workspaceRoot => {
+      const filePath = path.join(
+        workspaceRoot.uri.fsPath,
+        'force-app',
+        'main',
+        'default',
+        'lwc',
+        'demoLwcComponent',
+        'demoLwcComponent.html'
+      );
+
+      // Open an LWC HTML file
+      await vscode.workspace.openTextDocument(filePath);
+
+      assert.match(lwcExtension.isActive, true);
+    });
+  });
+
+  it('Language Server should not start for files that are not LWC', async () => {
+    const lwcExtension: vscode.Extension<any> = vscode.extensions.getExtension(
+      'salesforce.salesforcedx-vscode-lwc'
+    ) as vscode.Extension<any>;
+
+    if (!vscode.workspace.workspaceFolders) {
+      fail('Test requires workspace folders');
+      return;
+    }
+
+    vscode.workspace.workspaceFolders.forEach(async workspaceRoot => {
+      const filePath = path.join(workspaceRoot.uri.fsPath, 'sfdx-project.json');
+
+      // Open a non lwc file
+      await vscode.workspace.openTextDocument(filePath);
+
+      assert.match(lwcExtension.isActive, false);
+    });
+  });
+
+  it('Language Server should always start if the activation mode is set to "always"', async () => {
+    const activationMode = getActivationModeForWorkspace();
```

Comment:
Preserving the activation mode could also be handled using beforeEach/afterEach instead of try finally in each test.